.github/CHANGELOG.md

```diff
@@ -2,6 +2,52 @@
 
 <h3>New features since last release</h3>
 
+* The `Spy()` context manager has been added, which mocks selected methods 
+  in PennyLane without executing them, but instead registers statistics 
+  about the calls. 
+  
+  At the moment, the class only mocks the device's `execute` method and registers the number 
+  of times it has been called. As opposed to a device's `num_executions` attribute, this 
+  allows a sneak peak of the circuits evaluated in an arbitrary computation.
+  
+  The feature currently only works with devices that inherit from 
+  `QubitDevice` and in tape mode. 
+  [(#896)](https://github.com/PennyLaneAI/pennylane/pull/896) 
+  
+  ```python
+
+    import pennylane as qml
+    from pennylane.utils import Spy
+    qml.enable_tape()
+    
+    dev = qml.device("default.qubit", wires=1)
+    
+    @qml.qnode(dev)
+    def circuit(w):
+        qml.RX(w, wires=0)
+        qml.Hadamard(wires=0)
+        return qml.expval(qml.PauliZ(wires=0))
+       
+    with Spy() as spy:
```

Comment:
This is cute, the new name personifies it and makes it easy to remember the commands to call it.